tests/components/weatherflow/test_config_flow.py

```diff
@@ -0,0 +1,103 @@
+"""Test the IntelliFire config flow."""
+
+import asyncio
+from unittest.mock import AsyncMock, patch
+
+from homeassistant import config_entries
+from homeassistant.components.weatherflow.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_address_in_use(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_has_devices_error_address_in_use: AsyncMock,
+) -> None:
+    """Test the address in use error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["errors"] == {"base": "address_in_use"}
+
+
+async def test_cannot_connect(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_has_devices_error_listener
+) -> None:
+    """Test cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["errors"] == {"base": "cannot_connect"}
```

Comment:
We should complete all config flow test to reach an ABORT or CREATE_ENTRY state so we verify we can recover from error